core/src/main/java/org/geysermc/geyser/pack/ResourcePack.java

```diff
@@ -84,13 +86,18 @@ public static void loadPacks() {
         }
 
         List<Path> resourcePacks;
-        try {
-            resourcePacks = Files.walk(directory).collect(Collectors.toList());
+        try (Stream<Path> stream = Files.walk(directory)) {
+            resourcePacks = stream.collect(Collectors.toList());
         } catch (IOException e) {
             GeyserImpl.getInstance().getLogger().error("Could not list packs directory", e);
             return;
         }
 
+        Path skullResourcePack = SkullResourcePackManager.createResourcePack();
+        if (skullResourcePack != null) {
```

Comment:
Small issue I've found while testing this PR, right above this line there's a condition that checks if the pack directory exists and if it doesn't it returns (assuming that there are no packs to load) so the skull pack is not being generated if the packs directory doesn't exist.

Reply:
https://github.com/GeyserMC/Geyser/pull/3505/commits/f1bd04a71db4d65dcdca66d44dadaf62406efb56#diff-d9798739582712037f09ed06c6c3c8d4421a8a16722ac64cb35757243e5fae05R81